src/utils.js

```diff
@@ -76,29 +97,34 @@ export const indexOf = (array, value) => {
  */
 export const createReplacer = (replacer, circularReplacer) => {
   let cache = [];
+  let initd = 0;
 
+  const callable = typeof replacer === 'function';
   const getCircular = circularReplacer || getCircularValue;
 
   return function(key, value) {
-    if (cache.length) {
-      const locationOfThis = indexOf(cache, this);
+    let replacedValue = value;
+    if (initd) { // instead of read property of object every time
+      const locationOfThis = lastIndexOf(cache, this);
 
       if (~locationOfThis) {
-        cache = first(cache, locationOfThis + 1);
+        cache.length = locationOfThis + 1; // waste for create new array, we drop items at tail
       } else {
         cache[cache.length] = this;
       }
 
-      const locationOfValue = indexOf(cache, value);
+      const locationOfValue = lastIndexOf(cache, value);
 
       if (~locationOfValue) {
-        return getCircular.call(this, key, value, locationOfValue);
+        replacedValue = getCircular.call(this, key, value, locationOfValue);
+        // we dont return here, for call the replacer below even if circular was found
       }
     } else {
-      cache[cache.length] = value;
+      initd = 1;
+      cache[0] = value;
```

Comment:
i like the use of static assignment here, the contract is guaranteed so its safe and should be a smidge faster.